plugins/api/include/irods/plugins/api/switch_user_types.h

```diff
@@ -1,15 +1,82 @@
 #ifndef IRODS_SWITCH_USER_TYPES_H
 #define IRODS_SWITCH_USER_TYPES_H
 
+#include "irods/objInfo.h"
+
+/// The input data type used to invoke rc_switch_user.
+///
+/// \since 4.3.1
 typedef struct SwitchUserInput // NOLINT(modernize-use-using)
 {
-    // NOLINTBEGIN(modernize-avoid-c-arrays)
+    // NOLINTBEGIN(cppcoreguidelines-avoid-c-arrays, modernize-avoid-c-arrays)
+    // NOLINTBEGIN(cppcoreguidelines-avoid-magic-numbers, readability-magic-numbers)
+
+    /// The part of a fully-qualified iRODS username preceding the pound sign.
+    ///
+    /// The following requirements must be satisfied:
+    /// - The length must not exceed sizeof(UserInfo::userName) - 1 bytes
+    /// - It must be null-terminated
+    /// - It must be non-empty
+    /// - It must not include the pound sign
+    /// - It must not include the zone
+    ///
+    /// \since 4.3.1
     char username[64];
+
+    /// The part of a fully-qualified iRODS username following the pound sign.
+    ///
+    /// The following requirements must be satisfied:
+    /// - The length must not exceed sizeof(UserInfo::rodsZone) - 1 bytes
+    /// - It must be null-terminated
+    /// - It must be non-empty
+    /// - It must not include the pound sign
+    /// - It must not include the username
+    ///
+    /// \since 4.3.1
     char zone[64];
-    // NOLINTEND(modernize-avoid-c-arrays)
+
+    // NOLINTEND(cppcoreguidelines-avoid-magic-numbers, readability-magic-numbers)
+    // NOLINTEND(cppcoreguidelines-avoid-c-arrays, modernize-avoid-c-arrays)
+
+    /// TODO
+    ///
+    /// \since 4.3.1
+    struct KeyValPair options;
 } switchUserInp_t;
 
 // NOLINTNEXTLINE(cppcoreguidelines-macro-usage)
-#define SwitchUserInp_PI "str username[64]; str zone[64];"
+#define SwitchUserInp_PI "str username[64]; str zone[64]; struct KeyValPair_PI;"
+
+//
+// Options
+//
+
+/// TODO
+/// Influences how the server handles the proxy user.
+///
+/// If set to 1, the proxy user is switched too. This option will result in
+/// losing the ability to invoke rc_switch_user if this option is set to 1 and
+/// the user to switch to is not a \p rodsadmin. This option is not used when
+/// handling server-to-server connections.
+///
+/// \since 4.3.1
+#define KW_SWITCH_PROXY_USER            "switch_proxy_user"
```

Comment:
Should these keywords be moved to the rodsKeyWdDef.h header?

They are designed specifically for this API, but there's no reason they can't be used with others.

Thoughts anyone?

Reply:
defer

Reply:
These keywords belong in rodsKeyWdDef.h.

The documentation for each keyword will become part of the `options` member variable's documentation.

Reply:
done